src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1000 @@
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Determine the neighbors of a tile and push them onto the stack. If the 
+    !> tile neighbors have not been loaded, via add_tile, then they will be. 
+    !> Upon success, the neighbors of a tile will be pushed onto the stack and 
+    !> 0 will be returned.
+    !
+    !-----------------------------------------------------------------------
+    function mpas_geotile_mgr_push_neighbors(mgr, tile) result(ierr)
+
+        implicit none
+
+        class(mpas_geotile_mgr_type) :: mgr
+        type (mpas_geotile_type), pointer, intent(in) :: tile
+
+        integer :: ierr
+        type (mpas_geotile_type), pointer :: neighbor
+        real (kind=RKIND) :: lat
+        real (kind=RKIND) :: lon
+        integer :: xHash
+        integer :: yHash
+        integer :: nTileX
+        integer :: nTileY
+
+        ierr = 0
+
+        ! Up
+        ! Calculate the tile's hash coordinates
+        neighbor => null()
+        if (tile % hash_y + 1 > mgr % nTileY - 1) then 
+            xHash = modulo(tile % hash_x + (mgr % nTileX / 2), mgr % nTileX)
+            yHash = tile % hash_y
+        else
+            xHash = tile % hash_x
+            yHash = tile % hash_y + 1
+        endif
+        call mgr % hash_to_ll(xHash, yHash, lat, lon)
+
+        ierr = mgr % get_tile(lat, lon, neighbor)
+        if (ierr /= 0) then
+            call mpas_log_write("There was a problem getting the up tile", messageType=MPAS_LOG_ERR)
+            ierr = 1
+            return
+        endif
+        ierr = mgr % push_tile(neighbor)
+
+        ! Down
+        neighbor => null()
+        if (tile % hash_y - 1 < 0) then
+            xHash = modulo(tile % hash_x + (mgr % nTileX / 2), mgr % nTileX)
+            yHash = tile % hash_y
+        else
+            xHash = tile % hash_x + 1
+            yHash = tile % hash_y - 1
+        endif
+        call mgr % hash_to_ll(xHash, yHash, lat, lon)
+
+        ierr = mgr % get_tile(lat, lon, neighbor)
+        if (ierr /= 0) then
+            call mpas_log_write("There was a problem getting the down tile", messageType=MPAS_LOG_ERR)
+            ierr = 1
+            return
+        endif
+        ierr = mgr % push_tile(neighbor)
+
+        ! Right
+        neighbor => null()
+        if (tile % hash_x + 1 > mgr % nTileX - 1) then
+            yHash = 1
+            xHash = tile % hash_y
+        else
+            xHash = tile % hash_x + 1
+            yHash = tile % hash_y
+        endif
+        call mgr % hash_to_ll(xHash, yHash, lat, lon)
+
+        ierr = mgr % get_tile(lat, lon, neighbor)
+        if (ierr /= 0) then
+            call mpas_log_write("There was a problem getting the right tile", messageType=MPAS_LOG_ERR)
+            ierr = 1
+            return
+        endif
+        ierr = mgr % push_tile(neighbor)
+
+        ! Left
+        neighbor => null()
+        if (tile % hash_x -1 < 0) then
+            xHash = nTileX - 1
```

Comment:
The gfortran compiler seems correct in identifying `nTileX` as being used here before it has been defined:
```
mpas_geotile_manager.F:847:0:                                                                                                        
                                                                                                                                     
  847 |             xHash = nTileX - 1                                                                                               
      |                                                                                                                              
Warning: ‘ntilex’ may be used uninitialized in this function [-Wmaybe-uninitialized]
```

Reply:
nTileX and nTileY were both actually being defined here in the deceleration section, but not used as they are being replaced by `mgr % nTileX` and `mgr % ntileY`.